backend/apps/core/resilience.py

```
@@ -0,0 +1,89 @@
+        if state == "half_open":
+            logger.info(
+                f"Circuit breaker for service '{self.service_name}' closed after successful trial."
+            )
+            self.set_state("closed")
+        else:
+            cache.delete(self.failures_key)
+
+    def record_failure(self):
+        try:
+            failures = cache.get(self.failures_key, 0)
+        except Exception:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)